src/middlewared/middlewared/etc_files/local/ssh/config.py

```diff
@@ -24,6 +24,10 @@ def generate_ssh_config(middleware):
         if os.path.exists(os.path.join(SSH_CONFIG_PATH, f)):
             os.chmod(os.path.join(SSH_CONFIG_PATH, f), 0o600)
 
+    twofactor_auth = middleware.call_sync('auth.twofactor.config')
+    if any(not v for v in (twofactor_auth['enabled'], twofactor_auth['services']['ssh'])):
```

Comment:
Thats a over-engineered construct :P

`if not twofactor_auth['enabled'] or not twofactor_auth['services'].get('ssh'):`

looks much more direct, no? :P